crates/noirc_frontend/src/node_interner.rs

```diff
@@ -61,6 +61,13 @@ pub struct NodeInterner {
     // When resolving types, check against this map to see if a type alias is defined.
     type_aliases: Vec<TypeAliasType>,
 
+    // Trait map.
+    //
+    // Each struct definition is possibly shared across multiple type nodes.
+    // It is also mutated through the RefCell during name resolution to append
+    // methods from impls to the type.
+    traits: HashMap<TraitId, Shared<TraitType>>,
```

Comment:
```suggestion
    // Each trait definition is possibly shared across multiple type nodes.
    // It is also mutated through the RefCell during name resolution to append
    // methods from impls to the type.
    traits: HashMap<TraitId, Shared<TraitType>>,
```
We may be able to remove the `Shared` wrapper once traits are no longer types. We'd just lookup their methods as needed through the NodeInterner.

Reply:
nice